matchbox/provider_test.go

```diff
@@ -0,0 +1,251 @@
+package matchbox
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/coreos/matchbox/matchbox/rpc"
+	"github.com/coreos/matchbox/matchbox/server"
+	"github.com/coreos/matchbox/matchbox/storage"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+var providers = map[string]terraform.ResourceProvider{
+	"matchbox": Provider(),
+}
+
+func TestProvider(t *testing.T) {
+	if err := Provider().(*schema.Provider).InternalValidate(); err != nil {
+		t.Fatalf("err: %s", err)
+	}
+}
+
+type MatchboxFixtureServer struct {
+	Store    storage.Store
+	Listener net.Listener
+}
+
+func NewMatchboxFixtureServer(s storage.Store) *MatchboxFixtureServer {
+	l, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		panic(fmt.Errorf("failed to start listener: %v", err))
+	}
+
+	return &MatchboxFixtureServer{
+		Store:    s,
+		Listener: l,
+	}
+}
+
+func (s *MatchboxFixtureServer) Start() {
+	cfg, err := testServerTLSConfig()
+	if err != nil {
+		panic(fmt.Errorf("Invalid TLS credentials: %v", err))
+	}
+
+	srv := server.NewServer(&server.Config{s.Store})
```

Comment:
Also I need to keep the goroutine inside of the Start function otherwise we have a race condition when `AddProviderConfig` is called.